components/AddressForm.js

```diff
@@ -0,0 +1,444 @@
+                <StyledWarning>20 characters max</StyledWarning>
+              )}
+              {errors?.BillingFirstName?.type === 'pattern' && (
+                <StyledWarning>Alphabetical only</StyledWarning>
+              )}
+
+              <StyledInput
+                placeholder="Last Name*"
+                {...register('BillingLastName', {
+                  required: true,
+                  maxLength: 20,
+                  pattern: /^[A-Za-z]+$/i,
+                })}
+              />
+              {errors?.BillingLastName?.type === 'required' && (
+                <StyledWarning>required</StyledWarning>
+              )}
+              {errors?.BillingLastName?.type === 'maxLength' && (
+                <StyledWarning>20 characters max</StyledWarning>
+              )}
+              {errors?.BillingLastName?.type === 'pattern' && (
+                <StyledWarning>Alphabetical only</StyledWarning>
+              )}
+
+              <StyledInput
+                type="text"
+                placeholder="Email*"
+                {...register('BuyerEmail', {
+                  required: true,
+                  pattern: /^\S+@\S+$/i,
+                })}
+              />
+              {errors?.BuyerEmail?.type === 'required' && (
+                <StyledWarning>required</StyledWarning>
+              )}
+
+              {errors?.BuyerEmail?.type === 'pattern' && (
+                <StyledWarning>email not correct</StyledWarning>
+              )}
+              <StyledInput
+                placeholder="Company"
+                {...register('BillingCompany', {
+                  required: false,
+                  maxLength: 20,
+                })}
+              />
+              {errors?.BillingCompany?.type === 'maxLength' && (
+                <StyledWarning>20 characters max</StyledWarning>
+              )}
+
+              <StyledInput
+                placeholder="Street and number*"
+                {...register('BillingStreetAndNumber', {
+                  required: true,
+                  maxLength: 40,
+                })}
+              />
+              {errors?.BillingStreetAndNumber?.type === 'required' && (
+                <StyledWarning>required</StyledWarning>
+              )}
+              {errors?.BillingStreetAndNumber?.type === 'maxLength' && (
+                <StyledWarning>F40 characters max</StyledWarning>
+              )}
+
+              <StyledInput
+                placeholder="Optional line"
+                {...register('BillingOptionalLine', {
+                  required: false,
+                  maxLength: 20,
+                })}
+              />
+              {errors?.BillingOptionalLine?.type === 'maxLength' && (
+                <StyledWarning>too long</StyledWarning>
+              )}
+
+              <StyledInput
+                placeholder="ZIP*"
+                {...register('BillingZip', {
+                  required: true,
+                  maxLength: 10,
```

Comment:
Wieso hier 10 Zeichen? Und du solltest für die PLZ ein type=number verwenden.

Reply:
Nicht nur für Deutschland bestimmt. Manche EU Länder haben Buchstaben in der PLZ. Länge hebe ich nach Recherche sogar auf 12.